router/h2/src/main/scala/com/twitter/finagle/buoyant/h2/netty4/Netty4ClientDispatcher.scala

```diff
@@ -32,24 +38,33 @@ class Netty4ClientDispatcher(
 
   // TODO handle overflow
   private[this] val _id = new AtomicInteger(BaseStreamId)
-  private[this] def nextId() = _id.getAndAdd(2)
-  private[this] val streams = new ConcurrentHashMap[Int, Netty4ClientStreamTransport]
+  private[this] def nextId(): Int = {
+    val id = _id.getAndAdd(2)
+    if (id > MaxStreamId && closed.compareAndSet(false, true)) {
+      writer.goAwayProtocolError(Time.Top)
```

Comment:
Yep. There are no IDs left to use on this connection.  Need to start again on a fresh connection.